src/main/resources/frontend/src/stores/admin.ts

```diff
@@ -5,9 +5,18 @@ import {usersGet} from "@/services/adminService";
 
 export const useAdminStore = defineStore('admin', () => {
     const users = ref<User[]>([]);
+    const usersByNetId = ref<Record<string, User>>({});
 
     const updateUsers = async () => {
-        users.value = await usersGet();
+        const latestUsers = await usersGet();
+        users.value = latestUsers;
+
+        // Also store a cache of all the users by their netId
+        const byNetId: typeof usersByNetId = {};
+        for (let user of latestUsers) {
```

Comment:
TS demands some gymnastics here. `typeof usersByNetId` is actually `Ref<Record<string, User>>: value, [RefSymbol]` instead of just `Record<string, User>`

I'm not sure how to dig the inner type out of Refs in Vue, so using a common interface (or just duplicated type defs) would probably be a good solution

Reply:
Ah! Good point. I'm not familiar with `vite` to recognize the difference between their syntax and standard `TypeScript`. Thanks for the tips— I'll adjust it